### PR TITLE
allergies: Update to most recent spec

### PR DESCRIPTION
**Pull request by wolf99**
- Add missing test cases
- Update names of existing test cases

### CODE REVIEW

File: exercises/allergies/test/test_allergies.c [hunk 1]
```diff
@@ -10,34 +10,326 @@ void tearDown(void)
...
+
+   TEST_ASSERT_TRUE(is_allergic_to(ALLERGEN_EGGS, score));
+}
+
+static void test_allergic_to_something_but_not_eggs(void)
+{
+   TEST_IGNORE();
+   unsigned int score = 2;
+
+   TEST_ASSERT_FALSE(is_allergic_to(ALLERGEN_EGGS, score));
```
Comment (ryanplusplus) [inline exercises/allergies/test/test_allergies.c:41]: I know you're just copying the style of the existing tests, but how about this:

```suggestion
   TEST_ASSERT_FALSE(is_allergic_to(ALLERGEN_EGGS, 2));
```

I'm not sure how introducing a local helps to do anything except create separation between the input and the expected output.

**Comment by ryanplusplus** [comment]
Looks good overall, just one nitpick

File: exercises/allergies/test/test_allergies.c (hunk 1 above)
Comment (wolf99) [reply exercises/allergies/test/test_allergies.c]: Done

Comment (ryanplusplus) [reply exercises/allergies/test/test_allergies.c]: Thanks!

**Review by ryanplusplus — APPROVED** [review]

**Pull request merged**